packages/beacon-node/src/chain/chain.ts

```diff
@@ -892,8 +896,8 @@ export class BeaconChain implements IBeaconChain {
       state = await this.regen.getState(attHeadBlock.stateRoot, regenCaller);
     }
 
-    // resolve the promise to unblock other calls of the same epoch and dependent root
-    return this.shufflingCache.processState(state, attEpoch);
+    // should always be the current epoch of the active context so no need to await a result from the ShufflingCache
+    return state.epochCtx.getShufflingAtEpoch(attEpoch);
```

Comment:
we should still  call `processState()` here to cache the generated shuffling

Reply:
The should get cached during the regen process during epoch transitions.  Shufflings are not built by the EpochCache anymore, they are built in the ShufflingCache and referenced by the EpochCache

Reply:
got it, thanks 👍